design/one-pager-host-aware-stack-manager.md

```diff
@@ -0,0 +1,211 @@
+ the same Kubernetes API where controller pods deployed and custom resources exist which makes it mandatory to have a
+ dedicated Kubernetes Cluster per Crossplane installation. 
+
+## Goals
+
+- Deploy multiple isolated instances of Crossplane controllers on a single Kubernetes cluster (a.k.a. Host Cluster).
+- Make Stack Manager host aware so that it can deploy controller pods on Host cluster but watch and configure 
+ stack controllers to watch separate Kubernetes API instances for custom resources.
+
+## Proposal
+
+Stack Manager should support configuring different Kubernetes Apiservers for scheduling stack controllers and watching
+ custom resources. This will enable deploying multiple isolated Crossplane instances watching dedicated Kubernetes 
+ API instances (i.e. multiple API instances running in different namespaces) on a single Host Kubernetes cluster. 
+
+![host-aware-stack-manager](images/host-aware-stack-manager.png)
+
+## Required Changes
+
+- Stack Manager Changes: 
+    - It should be possible to configure Stack Manager against two different Kubernetes Apiservers. 
+    - Stack Manager should watch `stack` related CRs on Tenant Kubernetes Apiserver but should schedule stack 
+    installation jobs and controller deployments on Host Cluster.
+    - Stack Manager should configure stack controller pods in a way that they will communicate Tenant Kubernetes 
+    Apiservers instead of Host Cluster Apiserver.
+
+- Packaging Changes:
+    - It should be possible to deploy crossplane types and roles (CRDs and RBACs) separately from Crossplane and Stack
+    Manager pods.
+    
+### Stack Manager Changes
+
+#### Host Cluster Access
+
+Suggested way to access Kubernetes API from a pod is using [`in cluster config`](https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod).
+ However, for this proposal, we need to authenticate two different Kubernetes API's inside the stack manager pod and
+ we can only use `in cluster config` for one of them and will take a `kubeconfig` as parameter for the other.
+ 
+Option 1 - Using `in cluster config` for Host Cluster: Host Cluster already auto-mounts service account tokens which we
+ can use.
+ 
+Option 2 - Using `in cluster config` for Tenant Kubernetes: This will be consistent with other controllers (Crossplane 
+ and stack controllers).
+ 
+To leverage existing mechanisms to authenticate to host cluster, we will use Option 1. We will define RBAC rules for
+ stack manager which will be part of `crossplane-controllers` helm chart (see [packaging changes](#packaging-changes)) 
+ that will be deployed into host cluster. This way, we can explicitly declare permissions of stack manager on Host
+ Cluster.
+
+#### Stack Installation
+
+In case of a stack installation, Stack Manager should:
+
+1. Create install job on Host Cluster and read logs of install pod from Host.
+2. Deploy CRDs and RBAC on Tenant Kubernetes
```

Comment:
This list should include creating the `Stack` resource (and where that would reside)

Reply:
Since we moved away from owner based deletion (which didn't work anyway), a stack-manager must act on the resources of the stack. I assume the Stack resource lives in the host cluster for this reason.

Reply:
`Stack` resource also needs to be on Tenant Kubernetes instance. Please note, all our custom resources lives inside Tenant Instance and only workload related (i.e. generating a pod) resources (jobs, deployments) will live on host cluster.